migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java

```diff
@@ -86,8 +88,10 @@ public void checkpoint() {
         } else {
             if(waiting) {
                 synchronized(this) {
-                    ran=true;
-                    notify();
+                    if (waiting) {
```

Comment:
So if waiting is true before synchronized but false after, it should still notify?